spec/shared/clever_tap_client.rb

```diff
@@ -0,0 +1,36 @@
+shared_examples_for 'it calls success hook' do |times|
+  it "calls `on_successful_upload` #{times}" do
+    expect(success_proc).to receive(:call).send(times)
```

Comment:
it's hidden knowledge that you depend on having such a variable

Reply:
Is it ok if I do it like: ` it_behaves_like 'it calls hook', success, times`?